File: Week 2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Get user input
    string text = get_string("Text: ");

    // Count
    float letters = count_letters(text);
    float words = count_words(text);
    float sentences = count_sentences(text);

    // Calculate
    float l = letters / words * 100;
    float s = sentences / words * 100;
    int index = round(0.0588 * l - 0.296 * s - 15.8);

    if (index >= 16)
    {
        printf("Grade 16+\n");
        return 0;
    }
    if (index < 1)
    {
        printf("Before Grade 1\n");
        return 0;
    }
    printf("Grade %i\n", index);
}

int count_letters(string text)
{
    int count = 0;
    for (int i = 0, length = strlen(text); i < length; i++)
    {
        // Turn letter to lowercase
        char letter = tolower(text[i]);
        // Check if is a letter
        if (letter >= 'a' && letter <= 'z')
        {
            count++;
        }
    }
    return count;
}

int count_words(string text)
{
    int count = 1;
    for (int i = 0, length = strlen(text); i < length; i++)
    {
        char letter = text[i];
        if (letter == ' ')
        {
            count++;
        }
    }
    return count;
}

int count_sentences(string text)
{
    int count = 0;
    for (int i = 0, length = strlen(text); i < length; i++)
    {
        char letter = text[i];
        if (letter == '.' || letter == '?' || letter == '!')
        {
            count++;
        }
    }
    return count;
}

File: CS50X/Week 6/sentimental-readability/readability.py
from cs50 import get_string


# Take text from the user's input
text = get_string("Text: ")

sentences = 0
# 'words' must start from 1, because in 3 words there are only 2 spaces, in 7 words there are 6 spaces
words = 1
letters = 0

for i in text:
    if i in [".", "!", "?"]:
        sentences += 1
    elif i == " ":
        words += 1
    # If 'i' is alphabetic (a-z or A-Z)
    elif i.isalpha():
        letters += 1

# Average number of letters per 100 words in the text
L = (letters / words) * 100
# Average number of sentences per 100 words in the text
S = (sentences / words) * 100

# Coleman-Liau index
index = 0.0588 * L - 0.296 * S - 15.8
grade = round(index)

if grade < 1:
    print("Before Grade 1")
elif grade >= 16:
    print("Grade 16+")
else:
    print(f"Grade {grade}")
